File: python/sparklabs/batch/sales/rdd/distinct_customer.py
from sparklabs.batch.sales.rdd import utils


def run() -> None:
    distinct_customer_count = utils.orders.map(lambda order: order["customer_id"]).count()
    expensive_product = utils.products.takeOrdered(1, key=lambda product: -product["price"])

    print(f"Total distinct customers: {distinct_customer_count}")
    print("Most Expensive Product:")
    for row in expensive_product:
        print(row)


if __name__ == "__main__":
    run()

File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/rdd/DistinctCustomer.scala
package io.devorbit.sparklabs.batch.sales.rdd

object DistinctCustomer extends App {
  private val distinctCustomerCount = RddUtils.orders.map(_.customer_id).count()
  private val expensiveProduct = RddUtils.products.takeOrdered(1)(Ordering.by(product => -product.price))

  println(s"Total distinct customers: $distinctCustomerCount")
  println("Most Expensive Product:")
  expensiveProduct.foreach(println)
}
